pkg/cableengine/ipsec/ipsec.go

```diff
@@ -553,23 +571,34 @@ func runCharon(debug bool, logFile string) {
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 
+	var outputFile *os.File
 	if logFile != "" {
-		output, err := os.OpenFile(logFile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+		out, err := os.OpenFile(logFile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
 		if err != nil {
-			klog.Fatalf("Failed to log to file %s: %v", logFile, err)
+			return fmt.Errorf("Failed to open log file %s: %v", logFile, err)
 		}
-		defer output.Close()
-		cmd.Stdout = output
-		cmd.Stderr = output
+
+		cmd.Stdout = out
+		cmd.Stderr = out
+		outputFile = out
 	}
 
 	cmd.SysProcAttr = &syscall.SysProcAttr{
 		Pdeathsig: syscall.SIGTERM,
 	}
 
-	cmd.Start()
+	if err := cmd.Start(); err != nil {
+		// Note - Close handles nil receiver
+		outputFile.Close()
+		return fmt.Errorf("error starting the charon process wih args %v: %v", args, err)
+	}
+
+	go func() {
```

Comment:
I'm not sure this is equivalent to the original.

In the original the whole function was being parallelized (including the Start()) method that seems blocking.

In this case we only paralellize the Fatalf output. 

Shouldn't we paralellize the Start also, may be use a channel to return any error?

Reply:
That said It seems to be passing the E2E tests, but I'm not sure if this have some implication. Like: it will block the caller on Start ?

Reply:
Start() does not block awaiting completion:

    Start starts the specified command but does not wait for it to complete.
    The Wait method will return the exit code and release associated resources once the command exits.

Reply:
Oh right, I didn't see the Wait. That makes sense, you made the blocking part parallel again by use of the go + lambda function.